### ISSUE TITLE
The lightgbm use for time series

### ISSUE THREAD

**Issue by xtyi1997** [microsoft/LightGBM#4516]
I want to use lightgbm handle time series. The loss function is very important for me. I want to design it by myself.   
for example:
     def lossfunction(y_true, y_pred):
The shape of "y_true" is (n,)
How can I convert (n,) to  (n,5)?

**Comment by jameslamb** [comment]
Thanks for using LightGBM!

This issue does not currently have enough detail for maintainers to help you. Can you please share a reproducible example (sample code that we can run) which demonstrates what you're trying? If you can share some code with us and say "currently this fails, but I want it to do `<X>`", then we will be able to advise you on how to achieve what you're trying to achieve.

Also please see this example for using a custom objective function, if you have not already.

https://github.com/microsoft/LightGBM/blob/926526c838196a7497a85b6b8cf07657a88b69e6/examples/python-guide/advanced_example.py#L141-L146

That is a binary classification example but the interface for regression tasks like time series forecasting (assuming that is what you mean by "handle time series") is identical.

**Comment by xtyi1997** [comment]
Thanks your answer.
I want to add new cow.  
for example:
![image](https://user-images.githubusercontent.com/28750869/129433250-8a1e53bf-860c-4aac-ae80-6688a4387893.png)
The index_a is index of origin dataset,  (origin dataset will be preprocessed by shuffle)
so that I can compute  the distance  between normal sample and  fault sample.

**Comment by xtyi1997** [comment]
> Thanks for using LightGBM!
> 
> This issue does not currently have enough detail for maintainers to help you. Can you please share a reproducible example (sample code that we can run) which demonstrates what you're trying? If you can share some code with us and say "currently this fails, but I want it to do `<X>`", then we will be able to advise you on how to achieve what you're trying to achieve.
> 
> Also please see this example for using a custom objective function, if you have not already.
> 
> https://github.com/microsoft/LightGBM/blob/926526c838196a7497a85b6b8cf07657a88b69e6/examples/python-guide/advanced_example.py#L141-L146
> 
> That is a binary classification example but the interface for regression tasks like time series forecasting (assuming that is what you mean by "handle time series") is identical

Thanks your answer. 
The following is my question.
The shape of y_pred is (n,)
I want to convert to (n,5)(5 is random number)
![image](https://user-images.githubusercontent.com/28750869/129433507-e4dbe6d4-7920-477b-97b6-62d3cf71534a.png)
The index_a is index of origin dataset, (origin dataset will be preprocessed by shuffle)
so that I can compute the distance between normal sample and fault sample.

**Comment by jameslamb** [comment]
I apologize, but I don't understand the question.

Are you able to provide some code that demonstrates what you are trying to accomplish?
